docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+
+Notice a couple of things are different this time!
+
+It didn't ask you to fork the repository - that's because you already have a fork you're allowed to push to.
+
+You may also notice that if you click the "Create Pull Request" menu item, it brings you to a different page. It doesn't have a "Create Pull Request" button, it has a "View Pull Request" button. This is because if you already have a pull request open, when you push more changes, it will update your existing pull request.
+
+This is useful for responding to feedback you get on your homework. There are ways to open more than one pull request at a time (using something called branches), which we'll learn about in the future, but for now, one should do!
+
+## Getting feedback
+
+When you've made your pull request, our volunteers will be notified. They will look at your changes. When they are done, you will get an email. Here are some of the things they may do:
+* Make comments with suggestions, either about a particular bit of code, or about the whole pull request.
+* Add labels to the pull request (e.g. marking it as complete, or unfinished).
+
+If they give you suggestions, you should try to implement the suggestions, and push a new commit. If you're confused, struggling, or find them unclear, you can respond to the comment with a comment of your own, or ask on Slack.
+
+## History
+
+One useful features of Git is how it stores your commits. In Github Desktop, if you open the History tab, you can see a list of each commit that's been made in the repository, with the oldest at the bottom and the newest at the top.
+
+If you click on one of the commits, you'll see the changes that happened in it. This can be really useful to understand how the repository evolved to how it looks today. It can also help us to find out when bugs were introduced!
+
+You may notice that your commits all have messages like "Update README.md", whereas the ones before you started editing have different messages.
+
+These messages can be really useful to understand what a change did without having to read the whole thing. Let's try out an exercise to help us understand this:
+
+### Exercise (10 minutes)
+
+1. Clone https://github.com/illicitonion/git-log-example TODO: Move this repo into CodeYourFuture. TODO: Give instructions for how to clone a second repo (beacuse the screen is different from your first time).
+2. Have a read of the README.md file. See if you can find a problem in the file.
+3. Look through the history in Github Desktop. See if you can work out when and why the problem was introduced.
+4. Make a pull request fixing the problem.
+
+## Commit messages
+
+If you looked through the history of the Git repository, you may have been able to see where the problem came in just from the commit message, without having to look at the changes that were actually made. If the commit messages were all just "Update README.md", that would've been much harder!
+
```

Comment:
From @40thieves:

As mentioned above, IMO a good commit is more than just the message. It's about the scope of the changes. I don't believe there's a particularly good way of teaching this. It's the kind of thing that comes from experience.

Having said that, it may be good to discuss "commit early, commit often" as a good habit.

Reply:
I totally agree, I'm very much a "rewrite history until it's perfect" kind of person, but as you say, it comes from experience. Also worth noting that our homework (at least for JS1-3) is a really bad place to learn this - there are no really useful stages to decompose things into in our homeworks here :)